api/base/serializers.py

```diff
@@ -292,6 +292,8 @@ def __init__(self, source=None, **kwargs):
     def get_attribute(self, obj):
         user = self.context['request'].user
         auth = auth_core.Auth(user)
+        if 'view_only' in self.context['request'].query_params:
+            auth.private_key = self.context['request'].query_params['view_only']
```

Comment:
@brianjgeiger Normally the Auth object has the private_key set when a request is made with a view_only link, however in the case that it is made directly from the user like this, the private_key is always null, so calling has_anonymous_link(self.node, auth) will always be false with such an auth object. In order for that call to later succeed, we need to make sure the auth object has that information.